### PR TITLE
[Feature] Client Callbacks

### PR DESCRIPTION
**Pull request by ItsANoBrainer**
Creates support for the utilization of Client Callbacks.

client.lua
```lua
local password = 'hunter2'
QBCore.Functions.CreateClientCallback('example:client:GetClientPassword', function(cb, username)
    print('User requested password: ', username) -- prints 'IdrisSmells'
    cb(password) -- Sends the password back to the server
end)
```

server.lua
```lua
RegisterCommand('requestpassword', function(src)
    QBCore.Functions.TriggerClientCallback('example:client:GetClientPassword', src, function(password)
        print('We got that password: ', password) -- prints 'hunter2'
    end, 'IdrisSmells')
end, false)
```

### CODE REVIEW

File: server/functions.lua [hunk 1]
```diff
@@ -196,8 +196,15 @@ function PaycheckInterval()
...
+-- Callback Functions --
 
+-- Client Callback
+function QBCore.Functions.TriggerClientCallback(name, source, cb, ...)
+    QBCore.ClientCallbacks[name] = cb
+    TriggerClientEvent('QBCore:Client:TriggerClientCallback', source, name, ...)
+end
+
+-- Server Callback
 function QBCore.Functions.CreateCallback(name, cb)
```
Comment (uShifty) [inline server/functions.lua:208]: I think you should change the previous callback natives which would make it a breaking commit to match the naming schema. TriggerServerCallback/CreateServerCallback. Other than that I cannot wait for this to get merged

Comment (ItsANoBrainer) [reply server/functions.lua:208]: I thought about that as well. Seems like it seemed like too large of a breaking commit to do so but will see what everyone things.

**Comment by IdrisDose** [comment]
Looks fine.

However there was a discussion of changing the callback internals to have a unique id and name to prevent race conditions.

**Review by IdrisDose — APPROVED** [review]

**Review by xenfovn — APPROVED** [review]
NICE WORK

**Comment by GhzGarage** [comment]
Has conflicts

**Pull request merged**